src/arithmetic/fields.rs

```diff
@@ -130,14 +138,66 @@ pub trait FieldExt: ff::PrimeField + From<bool> + Ord + Group<Scalar = Self> {
     }
 }
 
+/// Tonelli-Shank's square-root algorithm for `p mod 16 = 1`.
+///
+/// https://eprint.iacr.org/2012/685.pdf (page 12, algorithm 5)
+///
+/// `tm1d2` should be set to `(t - 1) // 2`, where `t = (modulus - 1) >> F::S`.
+#[cfg(not(feature = "std"))]
+pub(crate) fn sqrt_tonelli_shank<F: ff::PrimeField, S: AsRef<[u64]>>(
```

Comment:
```suggestion
/// Tonelli-Shanks' square-root algorithm for `p mod 16 = 1`.
///
/// https://eprint.iacr.org/2012/685.pdf (page 12, algorithm 5)
///
/// `tm1d2` should be set to `(t - 1) // 2`, where `t = (modulus - 1) >> F::S`.
#[cfg(not(feature = "std"))]
pub(crate) fn sqrt_tonelli_shanks<F: ff::PrimeField, S: AsRef<[u64]>>(
```
[Shanks](https://en.wikipedia.org/wiki/Daniel_Shanks) is a name, not a possessive.

Reply:
Whoops, I propagated this back from the original impl removed in https://github.com/zcash/halo2/pull/120.